neqo-http3/src/connection_client.rs

```diff
@@ -5562,4 +5656,311 @@ mod tests {
             assert_closed(&client, &Error::HttpSettings);
         }
     }
+
+    #[test]
+    fn response_w_1xx() {
+        let (mut client, mut server, request_stream_id) = connect_and_send_request(true);
+
+        setup_server_side_encoder(&mut client, &mut server);
+
+        let headers1xx = vec![(String::from(":status"), String::from("101"))];
+
+        let encoded_headers = server
+            .encoder
+            .encode_header_block(&mut server.conn, &headers1xx, request_stream_id)
+            .unwrap();
+        let hframe = HFrame::Headers {
+            header_block: encoded_headers.to_vec(),
+        };
+        let mut d = Encoder::default();
+        hframe.encode(&mut d);
+        let headers200 = vec![
+            (String::from(":status"), String::from("200")),
+            (String::from("my-header"), String::from("my-header")),
+            (String::from("content-length"), String::from("3")),
+        ];
+
+        let encoded_headers = server
+            .encoder
+            .encode_header_block(&mut server.conn, &headers200, request_stream_id)
+            .unwrap();
+        let hframe = HFrame::Headers {
+            header_block: encoded_headers.to_vec(),
+        };
+        hframe.encode(&mut d);
+        // Send 1xx and 200 headers response.
+        server_send_response_and_exchange_packet(
+            &mut client,
+            &mut server,
+            request_stream_id,
+            &d,
+            false,
+        );
+
+        server_send_response_and_exchange_packet(
+            &mut client,
+            &mut server,
+            request_stream_id,
+            &[0x0, 0x0, 0x0],
+            false,
+        );
+
+        let mut events = client.events();
+        let (stream_id_1xx, interim_1xx, headers_1xx) = events
+            .find_map(|e| {
+                if let Http3ClientEvent::HeaderReady {
+                    stream_id,
+                    interim,
+                    headers,
+                    ..
+                } = e
+                {
+                    Some((stream_id, interim, headers))
+                } else {
+                    None
+                }
+            })
+            .unwrap();
+        assert_eq!(stream_id_1xx, request_stream_id);
+        assert!(interim_1xx);
+        assert_eq!(headers_1xx, headers1xx);
+
+        let (stream_id_200, interim_200, headers_200) = events
+            .find_map(|e| {
+                if let Http3ClientEvent::HeaderReady {
+                    stream_id,
+                    interim,
+                    headers,
+                    ..
+                } = e
+                {
+                    Some((stream_id, interim, headers))
+                } else {
+                    None
+                }
```

Comment:
Both of your closures here are the same.  You can use `filter_map()` instead and use `.next().unwrap()` to get each for testing.  As a bonus, you can add `assert!(iter.next().is_none())` at the end.

```rust
   let events= client.events().filter_map(|e| {
                if let Http3ClientEvent::HeaderReady {
                    stream_id,
                    interim,
                    headers,
                    ..
                } = e
                {
                    Some((stream_id, interim, headers))
                } else {
                    None
                }
);
  let (stream_id_1xx, interim_1xx, headers_1xx) = events.next().unwrap();
assert_eq!((stream_id_1xx, interim_1xx, headers_1xx), (request_stream_id, true, headers_1xx));
  let (stream_id_200, interim_200, headers_200) = events.next().unwrap();
assert_eq!((stream_id_200, interim_200, headers_200), (request_stream_id, false, headers_200));
assert!(events.next().is_none());
```